api/v1/storageclusterpeer_types.go

```diff
@@ -17,57 +17,37 @@ limitations under the License.
 package v1
 
 import (
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-type NamespacedName struct {
-	Name      string `json:"name"`
-	Namespace string `json:"namespace"`
-}
-
-// RemoteClusterSpec specifies the spec required for the remote cluster
-type RemoteClusterSpec struct {
-	// ApiEndpoint is the URI of the ODF api server
-	ApiEndpoint string `json:"apiEndpoint"`
-
-	// OnboardingTicket holds an identity information required by the local ODF cluster to onboard.
-	OnboardingTicket string `json:"onboardingTicket"`
 
-	// StorageClusterName holds the namespacedName of the Remote ODF Cluster
-	StorageClusterName NamespacedName `json:"storageClusterName"`
-}
+type StorageClusterPeerState string
 
-// LocalClusterSpec specifies the spec required for the local cluster
-type LocalClusterSpec struct {
-	// Name holds the name of the local ODF cluster
-	Name corev1.LocalObjectReference `json:"name"`
-}
+const (
+	StorageClusterPeerStatePending StorageClusterPeerState = "Pending"
+	StorageClusterPeerStatePeered  StorageClusterPeerState = "Peered"
+)
 
-// BlockPoolMirroringSpec enables setting up of mirroring for blockPools in the same namespace.
-type BlockPoolMirroringSpec struct {
-	// Selector is used to select blockPools by label
-	Selector metav1.LabelSelector `json:"selector"`
+type PeerInfo struct {
+	State                 StorageClusterPeerState `json:"state,omitempty"`
+	PeerStorageClusterUID string                  `json:"peerStorageClusterUID,omitempty"`
```

Comment:
You are already under the peer info section, there is no need to prefix the field with the word peer, as the field is already accessed as `Status.PeerInfo.StorageClusterUid`